extraction/src/main/scala/org/broadinstitute/monster/encode/extraction/ExtractionPipelineBuilder.scala

```diff
@@ -79,166 +110,152 @@ class ExtractionPipelineBuilder(getClient: () => EncodeClient)
+      targetField = "biosample.accession"
     )
 
     val replicates = extractLinkedEntities(
-      entityToExtract = EncodeEntity.Replicate,
-      matchingField = "accession",
-      linkingEntities = libraries,
-      linkedField = "library.accession"
+      sourceEntityType = EncodeEntity.Library,
+      sourceField = "accession",
+      sourceEntities = libraries,
+      targetEntityType = EncodeEntity.Replicate,
+      targetField = "library.accession"
     )
 
     val antibodies = extractLinkedEntities(
-      entityToExtract = EncodeEntity.AntibodyLot,
-      matchingField = "antibody",
-      linkingEntities = replicates
+      sourceEntityType = EncodeEntity.Replicate,
+      sourceField = "antibody",
+      sourceEntities = replicates,
+      targetEntityType = EncodeEntity.AntibodyLot,
+      targetField = "@id"
     )
 
     extractLinkedEntities(
-      entityToExtract = EncodeEntity.Target,
-      matchingField = "targets",
-      linkingEntities = antibodies
+      sourceEntityType = EncodeEntity.AntibodyLot,
+      sourceField = "targets",
+      sourceEntities = antibodies,
+      targetEntityType = EncodeEntity.Target,
+      targetField = "@id"
     )
 
     // partition the replicates stream into two separate SCollection[Msg]
     //passing in a new function to check to see if the experiment type
-    val (fcReplicate, expReplicate) = replicates.partition { replicate =>
-      ExtractionPipelineBuilder.isFunctionalCharacterizationReplicate(replicate)
-    }
+    val (fcReplicate, expReplicate) =
+      replicates.partition(isFunctionalCharacterizationReplicate)
 
     val experiments = extractLinkedEntities(
-      entityToExtract = EncodeEntity.Experiment,
-      matchingField = "experiment",
-      linkingEntities = expReplicate
+      sourceEntityType = EncodeEntity.Replicate,
+      sourceField = "experiment",
+      sourceEntities = expReplicate,
+      targetEntityType = EncodeEntity.Experiment,
+      targetField = "@id"
     )
 
     val fcExperiments = extractLinkedEntities(
-      entityToExtract = EncodeEntity.FunctionalCharacterizationExperiment,
-      matchingField = "experiment",
-      linkingEntities = fcReplicate
+      sourceEntityType = EncodeEntity.Replicate,
+      sourceField = "experiment",
+      sourceEntities = fcReplicate,
+      targetEntityType = EncodeEntity.FunctionalCharacterizationExperiment,
+      targetField = "@id"
     )
 
     val files = extractLinkedEntities(
-      entityToExtract = EncodeEntity.File,
-      matchingField = "@id",
-      linkingEntities = SCollection.unionAll(List(experiments, fcExperiments)),
-      linkedField = "dataset"
+      // Fudging a little, but it's only used for logging so it's ok.
+      sourceEntityType = EncodeEntity.Experiment,
+      sourceField = "files",
+      sourceEntities = SCollection.unionAll(List(experiments, fcExperiments)),
+      targetEntityType = EncodeEntity.File,
+      targetField = "@id"
```

Comment:
This was the original motivation: instead of pulling out the `@id`s of experiments and joining against `dataset` (returning around an order of magnitude more things than `batchSize` asked for), we pull out `files` and join against `@id`.